MobileDevProject/ViewController.swift

```diff
@@ -107,12 +99,27 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
     
     func locationManager(_ manager: CLLocationManager, didRangeBeacons beacons: [CLBeacon], in region: CLBeaconRegion) {
         if beacons.count > 0 {
-            for beacon in beaconInfo {
-                if (beacon.value == beacons[0].minor) {
-                    beacon.button.backgroundColor = UIColor.blue
-                    nearestBeaconCoordinate = beacon.coordinate
+            beaconsArray = beacons
+            for myBeacon in beaconInfo {
+                if (myBeacon.value == beacons[0].minor) {
+                    var buttonColour: UIColor
+                    let greenAmount = (255 - (CGFloat(beacons[0].accuracy) * 40))
+                    buttonColour = UIColor.init(red: 0, green: CGFloat(greenAmount/255), blue: 0, alpha: 1)
+                    myBeacon.button.backgroundColor = buttonColour
+                    nearestBeacon = myBeacon
                 } else {
-                    beacon.button.backgroundColor = UIColor.red
+                    myBeacon.button.backgroundColor = UIColor.red
+                }
+                
+                if(isNavigating){
```

Comment:
`if(isNavigating){` => `if (isNavigating) {` : easier to look and we should be consistent.